File: sdks/csharp/src/NovaBilling/Subscriptions/Requests/ListSubscriptionsRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record ListSubscriptionsRequest
{
    /// <summary>
    /// Filter by status (ACTIVE, TRIALING, PAUSED, CANCELED)
    /// </summary>
    [JsonIgnore]
    public string? Status { get; set; }

    /// <summary>
    /// Filter by customer ID
    /// </summary>
    [JsonIgnore]
    public string? CustomerId { get; set; }

    /// <summary>
    /// Filter by plan ID
    /// </summary>
    [JsonIgnore]
    public string? PlanId { get; set; }

    [JsonIgnore]
    public double? Page { get; set; }

    [JsonIgnore]
    public double? Limit { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/Subscriptions/Requests/ListSubscriptionsRequest.php
<?php

namespace NovaBilling\Subscriptions\Requests;

use NovaBilling\Core\Json\JsonSerializableType;

class ListSubscriptionsRequest extends JsonSerializableType
{
    /**
     * @var ?string $status Filter by status (ACTIVE, TRIALING, PAUSED, CANCELED)
     */
    public ?string $status;

    /**
     * @var ?string $customerId Filter by customer ID
     */
    public ?string $customerId;

    /**
     * @var ?string $planId Filter by plan ID
     */
    public ?string $planId;

    /**
     * @var ?float $page
     */
    public ?float $page;

    /**
     * @var ?float $limit
     */
    public ?float $limit;

    /**
     * @param array{
     *   status?: ?string,
     *   customerId?: ?string,
     *   planId?: ?string,
     *   page?: ?float,
     *   limit?: ?float,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->status = $values['status'] ?? null;
        $this->customerId = $values['customerId'] ?? null;
        $this->planId = $values['planId'] ?? null;
        $this->page = $values['page'] ?? null;
        $this->limit = $values['limit'] ?? null;
    }
}
